syncer/sharding-meta/shardmeta_test.go

```diff
@@ -0,0 +1,265 @@
+	c.Assert(meta.GetActiveDDLItem(table2), check.DeepEquals, items[4])
+	c.Assert(meta.GetActiveDDLItem(table3), check.DeepEquals, items[7])
+	c.Assert(meta.InSequenceSharding(), check.IsTrue)
+	pos, err = meta.ActiveDDLFirstPos()
+	c.Assert(err, check.IsNil)
+	c.Assert(*pos, check.DeepEquals, items[1].FirstPos)
+
+	// find synced in shrading group, and call ShardingMeta.ResolveShardingDDL
+	c.Assert(meta.ResolveShardingDDL(), check.IsFalse)
+
+	c.Assert(meta.GetActiveDDLItem(table1), check.DeepEquals, items[2])
+	c.Assert(meta.GetActiveDDLItem(table2), check.DeepEquals, items[5])
+	c.Assert(meta.GetActiveDDLItem(table3), check.IsNil)
+	c.Assert(meta.InSequenceSharding(), check.IsTrue)
+	pos, err = meta.ActiveDDLFirstPos()
+	c.Assert(err, check.IsNil)
+	c.Assert(*pos, check.DeepEquals, items[2].FirstPos)
+
+	sqls, args = meta.FlushData(metaSchema, metaTable, sourceID, tableID)
+	c.Assert(sqls, check.HasLen, 4)
+	c.Assert(args, check.HasLen, 4)
+	for _, stmt := range sqls {
+		c.Assert(stmt, check.Matches, fmt.Sprintf("INSERT INTO .*"))
+	}
+	for _, arg := range args {
+		c.Assert(arg, check.HasLen, 8)
+		c.Assert(arg[3], check.Equals, 2)
+	}
+
+	// 3rd round sharding DDL sync
+	for i := 0; i < 9; i++ {
+		if i%3 != 2 {
+			continue
+		}
+		active, err = meta.AddItem(items[i])
+		c.Assert(err, check.IsNil)
+		if i%3 == 2 {
+			c.Assert(active, check.IsTrue)
+		} else {
+			c.Assert(active, check.IsFalse)
+		}
+	}
+	c.Assert(meta.GetActiveDDLItem(table1), check.DeepEquals, items[2])
+	c.Assert(meta.GetActiveDDLItem(table2), check.DeepEquals, items[5])
+	c.Assert(meta.GetActiveDDLItem(table3), check.DeepEquals, items[8])
+	c.Assert(meta.InSequenceSharding(), check.IsTrue)
+	pos, err = meta.ActiveDDLFirstPos()
+	c.Assert(err, check.IsNil)
+	c.Assert(*pos, check.DeepEquals, items[2].FirstPos)
+
+	// find synced in shrading group, and call ShardingMeta.ResolveShardingDDL
+	c.Assert(meta.ResolveShardingDDL(), check.IsTrue)
+
+	c.Assert(meta.GetActiveDDLItem(table1), check.IsNil)
+	c.Assert(meta.GetActiveDDLItem(table2), check.IsNil)
+	c.Assert(meta.GetActiveDDLItem(table3), check.IsNil)
+	c.Assert(meta.InSequenceSharding(), check.IsFalse)
+	pos, err = meta.ActiveDDLFirstPos()
+	c.Assert(err, check.ErrorMatches, fmt.Sprintf("activeIdx %d larger than global DDLItems:.*", meta.ActiveIdx()))
+
+	sqls, args = meta.FlushData(metaSchema, metaTable, sourceID, tableID)
+	c.Assert(sqls, check.HasLen, 1)
+	c.Assert(args, check.HasLen, 1)
+	c.Assert(sqls[0], check.Matches, fmt.Sprintf("DELETE FROM .*"))
+	c.Assert(args[0], check.DeepEquals, []interface{}{sourceID, tableID})
+}
+
+func (t *testShardMetaSuite) TestShardingMetaWrongSequence(c *check.C) {
+	var (
+		active   bool
+		err      error
+		filename = "mysql-bin.000001"
+		table1   = "table1"
+		table2   = "table2"
+		meta     = NewShardingMeta()
+		items    = []*DDLItem{
+			NewDDLItem(mysql.Position{filename, 1000}, []string{"ddl1"}, table1),
```

Comment:
can we add some test case with more than one ddl in DDLItem's DDLS?

Reply:
added in https://github.com/pingcap/dm/pull/177/commits/790059ae816b17d156a68a18237cda7ffc533e2b